ImageDetection/ImageDetection/blob_detector.py

```diff
@@ -0,0 +1,132 @@
+import cv2
+import numpy
+import PIL.ImageOps
+from matplotlib import pyplot
+from PIL import Image
+from SyntheticDataset2 import *
+from .image_detection_settings import ImageDetectionSettings
+
+class BlobDetector(object):
+    """
+    The following code and descriptions are adapted from OpenCV:
+    """
+    def __init__(self, image_path, target_size_range_in_pixels):
+        """
+        Read image.
+        """
+        self.image_path = image_path
+        self.target_size_range_in_pixels = target_size_range_in_pixels
+        self.raw_image = Image.open(image_path)
+        self.minimum_area = target_size_range_in_pixels[0]
+        self.maximum_area = (target_size_range_in_pixels[1])**1.5
+
+    def detect_blobs(self):
+        """
+        The algorithm of SimpleBlobDetector follows the following steps:
+
+        Thresholding: Convert the source images to several binary images by
+                      thresholding the source image with thresholds starting at
+                      minThreshold. These thresholds are incremented by
+                      thresholdStep until maxThreshold. So the first threshold
+                      is minThreshold, the second is
+                      minThreshold + thresholdStep, the third is
+                      minThreshold + 2 x thresholdStep, and so on.
+
+        Grouping: In each binary image, connected white pixels are grouped
+                  together. Let us call these binary blobs.
+
+        Merging: The centers of the binary blobs in the binary images are
+                 computed, and blobs located closer than minDistBetweenBlobs are
+                 merged.
+
+        Center & Radius Calculation: The centers and radii of the new merged
+                                     blobs are computed and returned.
+        """
+
+        """
+        Set up the detector with default parameters.
+        """
```

Comment:
Single-line comments should begin with a `#` and not use the `"""` comment begin/close.